### PR TITLE
Clean up ContentViewCategory

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This PR makes some clean up to ContentViewCategory classes:

1. Use namespace classes instead of it's alias
2. Move $dispatcher = JEventDispatcher::getInstance(); out of the loop, so we get a bit performance improvement
3. Remove code to set view layout as it is set on parent class code already https://github.com/joomla/joomla-cms/blob/staging/libraries/src/MVC/View/CategoryView.php#L212-L223

4. Remove some un-used variables (like $id, $pathway), reuse-existing variable (no need for $menu = $menu->getActive() when we have $active variable keep active menu item before)

### Testing Instructions
Code review should be enough. For human testing, please create a menu item to display articles from a category (Category Blog layout menu option for example), check and make sure articles still being displayed like before

### CODE REVIEW

File: components/com_content/views/category/view.html.php [hunk 1]
```diff
@@ -79,7 +81,8 @@ public function display($tpl = null)
 		$numLinks   = $params->def('num_links', 4);
 		$this->vote = JPluginHelper::isEnabled('content', 'vote');
 
-		JPluginHelper::importPlugin('content');
+		PluginHelper::importPlugin('content');
```
Comment (Quy) [inline components/com_content/views/category/view.html.php]: Line 82 has a prefix `J`. This one does not. Should they be the same and with or without `J`?

Comment (dgrammatiko) [reply components/com_content/views/category/view.html.php]: Without check line 13

Comment (joomdonation) [reply components/com_content/views/category/view.html.php]: Without J as @dgt41 pointed out. I updated the code. If #19234 and #19239 are accepted, I will still need to make another PR to complete clean up code in this class.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 02f1b743044d961b9899553045f375895e36d565<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19237">issues.joomla.org/tracker/joomla-cms/19237</a>.</sub>

**Comment by joomdonation** [comment]
Ouch

**Pull request closed without merging**